test/workspace_test.dart

```diff
@@ -1416,6 +1416,79 @@ Consider removing one of the overrides.''',
       ),
     );
   });
+
+  test('workspace list', () async {
+    await dir(appPath, [
+      libPubspec(
+        'myapp',
+        '1.2.3',
+        extras: {
+          'workspace': ['pkgs/a'],
+        },
+        sdk: '^3.5.0',
+      ),
+      dir('pkgs', [
+        dir('a', [
+          libPubspec(
+            'a',
+            '1.1.1',
+            resolutionWorkspace: true,
+            extras: {
+              'workspace': ['b'],
+            },
+          ),
+          dir('b', [
+            libPubspec(
+              'b',
+              '1.2.2',
+              resolutionWorkspace: true,
+            ),
+          ]),
+        ]),
+      ]),
+    ]).create();
+    await runPub(
+      args: ['workspace', 'list'],
+      environment: {'_PUB_TEST_SDK_VERSION': '3.5.0'},
+      output: '''
+myapp: .
+a: pkgs/a
+b: pkgs/a/b
+''',
+    );
+    await runPub(
+      args: ['workspace', 'list'],
+      environment: {'_PUB_TEST_SDK_VERSION': '3.5.0'},
+      workingDirectory: p.join(sandbox, appPath, 'pkgs'),
+      output: '''
+myapp: ..
+a: a
+b: a/b
```

Comment:
Should we consider:
 * Some right pad?
 * Maybe tabel headers too (possibly bold, like we do in `dart pub outdated`)
 * Ending paths in `/` to indicate that they are directories?

**Example**
```
Package:                Path:
_                       ./
a                       a/
b                       b/
```

In the future, I could imagine a `resolution` column too:
```
Package:                Path:               Resolution:
_                       ./                  workspace
a                       a/                  workspace
b                       b/                  local
```

Reply:
Done